File: 11_march_madness/main.go
package main

import (
	"fmt"
	"math/big"
)

func main() {
	gameCount := gameCountByTeamCount(64)
	fmt.Println("Games:", gameCount)
	combinations := big.NewInt(1)
	big2 := big.NewInt(2)
	for i := 0; i < gameCount; i++ {
		combinations.Mul(combinations, big2)
	}

	fmt.Println("Combinations:", combinations)
}

func gameCountByTeamCount(teamCount int) int {
	return teamCount - 1
}

File: 11_march_madness/main.rb
def game_count_by_team_count(team_count)
  if team_count > 2
    game_count = team_count / 2
    game_count + game_count_by_team_count(game_count)
  elsif team_count == 2
    1
  else
    raise "Invalid team_count"
  end
end

game_count = game_count_by_team_count 64
puts "Games: #{game_count}"

combinations = 2**game_count
puts "Combinations: #{combinations}"
